### PR TITLE
[Process] throw when PhpProcess::fromShellCommandLine() is used

### LINKED ISSUE #35637: [Process] PhpProcess constructor arguments
**Symfony version(s) affected**: master

**Description**  
Using the `PhpProcess::fromShellCommandline()` method and passing a simple string `echo Symfony`, it seems that PhpProcess constructor requires a string, by default, the `fromShellCommandline` implementation depends on the `Process` one which requires an array as the first argument.

**How to reproduce**  
Just try to instantiante `PhpProcess::fromShellCommandline()` with a string `echo Symfony`. 

**Possible Solution**  

#35641

**Additional context**  
As far as I can see, it seems that the error occurs here: 

```php
$process = new static([], $cwd, $env, $input, $timeout);
$process->commandline = $command;

return $process;
```

`PhpProcess` receive an array as the first argument 🤔

### PR DESCRIPTION
**Pull request by Guikingone**
| Q             | A
| ------------- | ---
| Branch?       | 4.4 
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #35637
| License       | MIT
| Doc PR        | None

Close #35638 

Final PR (rebased and tests added)

### CODE REVIEW

File: src/Symfony/Component/Process/PhpProcess.php [hunk 1]
```diff
@@ -49,6 +50,14 @@ public function __construct(string $script, string $cwd = null, array $env = nul
         parent::__construct($php, $cwd, $env, $script, $timeout);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public static function fromShellCommandline(string $command, string $cwd = null, array $env = null, $input = null, ?float $timeout = 60)
+    {
+        throw new LogicException(sprintf('The "%s" cannot be called when using "%s"', __METHOD__, self::class));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/PhpProcess.php]: missing final dot :)

Comment (nicolas-grekas) [reply src/Symfony/Component/Process/PhpProcess.php]: missing work "method" also, and brackets

Comment (Guikingone) [reply src/Symfony/Component/Process/PhpProcess.php]: Improved 🙂

**Comment by Guikingone** [comment]
Maybe it could be a good idea to submit a PR on the doc to add a warning about calling this method? 🤔

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Guikingone.

**Pull request merged**